pkg/portal/libstripe/service.go

```diff
@@ -338,30 +338,43 @@ func (s *Service) intersectPlanNames(plans []*model.Plan, products []*stripe.Pro
 		setA[plan.Name] = struct{}{}
 	}
 
-	setB := make(map[string]struct{})
+	productPlanSet := make(map[string]planWithVersion)
 	for _, product := range products {
-		planName, ok := product.Metadata[MetadataKeyPlanName]
-		if ok && planName != "" {
-			setB[planName] = struct{}{}
+		planName, planNameOk := product.Metadata[MetadataKeyPlanName]
+		version := product.Metadata[MetadataKeyVersion]
+		if planNameOk && planName != "" {
+			planWithVersion := planWithVersion{
+				PlanName: planName,
+				Version:  version,
+			}
+			productPlanSet[planName] = planWithVersion
 		}
 	}
 
-	intersection := make(map[string]struct{})
+	intersection := setutil.Set[string]{}
 
 	for a := range setA {
-		_, ok := setB[a]
+		_, ok := productPlanSet[a]
 		if ok {
 			intersection[a] = struct{}{}
 		}
 	}
 
-	return intersection
+	intersactedProductPlanWithVersions := []planWithVersion{}
+	for _, planName := range intersection.Keys() {
+		planWithVersion := productPlanSet[planName]
```

Comment:
Old plans do not have version, therefore we treat them as version="" which is the zero value. So no I intentionally do not check ok.

Reply:
I mean the final return value could contain a zero value in the slice. I think it is not expected?

Reply:
It is expected, because zero value of string is empty string "", and plans without version should be treated as empty string version and therefore they are still in the same group.

Reply:
Oh I understand what you mean. Yes I think I should check if the plan exist in productPlanSet or not